### PR TITLE
chore: Fix linter findings for errorlint (part3)

### PR DESCRIPTION
**Pull request by zak-pawel**
Address findings for [errorlint](https://github.com/polyfloyd/go-errorlint) - finds code that can cause problems with the error wrapping scheme introduced in Go 1.13.

It is only part of the bigger job.
After all findings in whole project are handled, we can enable `errorlint` linter to guard this.

Following findings in `plugins/outputs` package were fixed:
```
plugins/outputs/amon/amon.go:86:57                                          errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/amon/amon.go:90:58                                          errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/amon/amon.go:96:50                                          errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/amon/amon.go:116:69                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/amqp/amqp.go:164:19                                         errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/amqp/client.go:73:55                                        errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/amqp/client.go:113:53                                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/amqp/client.go:142:19                                       errorlint  comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/azure_data_explorer/azure_data_explorer.go:193:74           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/azure_data_explorer/azure_data_explorer.go:198:77           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/azure_monitor/azure_monitor.go:190:59                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/azure_monitor/azure_monitor.go:326:71                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/azure_monitor/azure_monitor.go:331:6                        errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/azure_monitor/azure_monitor.go:331:6                        errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/cloud_pubsub/cloud_pubsub.go:125:61                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/datadog/datadog.go:134:57                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/datadog/datadog.go:195:88                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/dynatrace/dynatrace.go:137:53                               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/dynatrace/dynatrace.go:150:63                               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/dynatrace/dynatrace.go:168:62                               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/elasticsearch/elasticsearch.go:172:47                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/elasticsearch/elasticsearch.go:208:63                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/elasticsearch/elasticsearch.go:313:72                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/elasticsearch/elasticsearch.go:341:106                      errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/elasticsearch/elasticsearch.go:373:95                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/exec/exec.go:103:6                                          errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/exec/exec.go:117:63                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/exec/exec.go:120:51                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/exec/exec.go:167:7                                          errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/execd/execd.go:92:50                                        errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/execd/execd_test.go:165:7                                   errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/execd/execd_test.go:168:33                                  errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/file/file.go:100:58                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/graphite/graphite.go:141:5                                  errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/graphite/graphite.go:152:14                                 errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/graphite/graphite.go:187:49                                 errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/graylog/graylog.go:462:63                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/groundwork/groundwork.go:118:45                             errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/groundwork/groundwork.go:126:46                             errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/http/http.go:242:71                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/http/http.go:265:68                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/http/http.go:270:61                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/http.go:333:52                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/http.go:341:52                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/http.go:347:45                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/http.go:464:61                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/influxdb.go:90:55                                  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/influxdb.go:97:72                                  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/influxdb.go:148:22                                 errorlint  type switch on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/influxdb/influxdb.go:182:73                                 errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/influxdb.go:218:74                                 errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb/udp.go:88:63                                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb_v2/http.go:190:18                                  errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/influxdb_v2/http.go:222:19                                  errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/influxdb_v2/influxdb_v2.go:68:55                            errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb_v2/influxdb_v2.go:75:72                            errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/influxdb_v2/influxdb_v2.go:147:74                           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/instrumental/instrumental.go:82:74                          errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/instrumental/instrumental.go:151:6                          errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/kafka/kafka.go:149:63                                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/kafka/kafka.go:212:60                                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/kafka/kafka.go:224:18                                       errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/kafka/kafka.go:226:8                                        errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/kafka/kafka.go:230:8                                        errorlint  comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error
plugins/outputs/librato/librato.go:132:54                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/librato/librato.go:142:58                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/librato/librato.go:162:50                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/librato/librato.go:209:73                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/logzio/logzio.go:100:54                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/logzio/logzio.go:105:56                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/logzio/logzio.go:111:49                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/logzio/logzio.go:125:58                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/logzio/logzio.go:132:50                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/mongodb/mongodb.go:31:62                                    errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/mongodb/mongodb.go:36:62                                    errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/mongodb/mongodb.go:172:69                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/mongodb/mongodb.go:183:46                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/mongodb/mongodb.go:187:85                                   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/mqtt/mqtt.go:129:61                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/mqtt/mqtt.go:142:60                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/nats/nats.go:111:57                                         errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/newrelic/newrelic.go:72:57                                  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/nsq/nsq.go:66:57                                            errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/opentsdb/opentsdb.go:70:54                                  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/opentsdb/opentsdb.go:76:68                                  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/opentsdb/opentsdb.go:80:57                                  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/opentsdb/opentsdb.go:93:54                                  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/opentsdb/opentsdb.go:192:60                                 errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/postgresql/postgresql.go:357:16                             errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/postgresql/postgresql.go:403:93                             errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/postgresql/table_manager.go:388:57                          errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/prometheus_client/prometheus_client.go:116:52               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/redistimeseries/redistimeseries.go:83:51                    errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/riemann/riemann.go:73:62                                    errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/riemann/riemann.go:86:59                                    errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/riemann_legacy/riemann_legacy.go:67:69                      errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/riemann_legacy/riemann_legacy.go:80:90                      errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/sensu/sensu.go:354:56                                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/socket_writer/socket_writer.go:122:18                       errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/socket_writer/socket_writer.go:126:49                       errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/sql/sql.go:245:43                                           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/sql/sql.go:249:45                                           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/sql/sql.go:255:47                                           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/sql/sql.go:259:44                                           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/sql/sql.go:264:47                                           errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/stomp/stomp.go:87:51                                        errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/syslog/syslog.go:123:21                                     errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/syslog/syslog.go:126:49                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/timestream/timestream.go:290:51                             errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/timestream/timestream.go:305:51                             errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/timestream/timestream.go:311:51                             errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/timestream/timestream.go:318:51                             errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/timestream/timestream.go:371:15                             errorlint  type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
plugins/outputs/warp10/warp10.go:125:68                                     errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/wavefront/wavefront.go:129:55                               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/websocket/websocket.go:71:54                                errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/websocket/websocket.go:76:49                                errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/websocket/websocket.go:88:62                                errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/websocket/websocket.go:100:39                               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/websocket/websocket.go:165:58                               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/websocket/websocket.go:176:56                               errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:161:56   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:189:91   errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:205:101  errorlint  non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

```

### CODE REVIEW

**Review by powersj — APPROVED** [review]

File: plugins/outputs/newrelic/newrelic.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func (nr *NewRelic) Connect() error {
 			}
 		})
 	if err != nil {
-		return fmt.Errorf("unable to connect to newrelic %v", err)
+		return fmt.Errorf("unable to connect to newrelic %w", err)
```
Comment (Hipska) [inline plugins/outputs/newrelic/newrelic.go:72]:
```suggestion
		return fmt.Errorf("unable to connect to newrelic: %w", err)
```

File: plugins/outputs/opentsdb/opentsdb.go [hunk 2]
```diff
@@ -187,9 +187,9 @@ func (o *OpenTSDB) WriteTelnet(metrics []telegraf.Metric, u *url.URL) error {
 				sanitize(fmt.Sprintf("%s%s%s%s", o.Prefix, m.Name(), o.Separator, fieldName)),
 				now, metricValue, tags)
 
-			_, err := connection.Write([]byte(messageLine))
+			_, err = connection.Write([]byte(messageLine))
 			if err != nil {
-				return fmt.Errorf("OpenTSDB: Telnet writing error %s", err.Error())
+				return fmt.Errorf("OpenTSDB: Telnet writing error %w", err)
```
Comment (Hipska) [inline plugins/outputs/opentsdb/opentsdb.go:192]:
```suggestion
				return fmt.Errorf("telnet writing error: %w", err)
```

File: plugins/outputs/postgresql/postgresql.go [hunk 3]
```diff
@@ -398,9 +399,9 @@ func (p *Postgresql) writeMetricsFromMeasure(ctx context.Context, db dbh, tableS
 	}
 
 	if p.TagsAsForeignKeys {
-		if err := p.writeTagTable(ctx, db, tableSource); err != nil {
+		if err = p.writeTagTable(ctx, db, tableSource); err != nil {
 			if p.ForeignTagConstraint {
-				return fmt.Errorf("writing to tag table '%s': %s", tableSource.Name()+p.TagTableSuffix, err)
+				return fmt.Errorf("writing to tag table '%s': %w", tableSource.Name()+p.TagTableSuffix, err)
```
Comment (Hipska) [inline plugins/outputs/postgresql/postgresql.go:404]:
```suggestion
				return fmt.Errorf("writing to tag table %q: %w", tableSource.Name()+p.TagTableSuffix, err)
```

File: plugins/outputs/riemann_legacy/riemann_legacy.go [hunk 4]
```diff
@@ -64,7 +64,7 @@ func (r *Riemann) Write(metrics []telegraf.Metric) error {
 	if r.client == nil {
 		err := r.Connect()
 		if err != nil {
-			return fmt.Errorf("failed to (re)connect to Riemann, error: %s", err)
+			return fmt.Errorf("failed to (re)connect to Riemann, error: %w", err)
```
Comment (Hipska) [inline plugins/outputs/riemann_legacy/riemann_legacy.go:67]:
```suggestion
			return fmt.Errorf("failed to (re)connect to Riemann: %w", err)
```

File: plugins/outputs/riemann_legacy/riemann_legacy.go [hunk 5]
```diff
@@ -77,7 +77,7 @@ func (r *Riemann) Write(metrics []telegraf.Metric) error {
 	var senderr = r.client.SendMulti(events)
 	if senderr != nil {
 		r.Close() //nolint:revive // There is another error which will be returned here
-		return fmt.Errorf("failed to send riemann message (will try to reconnect), error: %s", senderr)
+		return fmt.Errorf("failed to send riemann message (will try to reconnect), error: %w", senderr)
```
Comment (Hipska) [inline plugins/outputs/riemann_legacy/riemann_legacy.go:80]:
```suggestion
		return fmt.Errorf("failed to send riemann message (will try to reconnect): %w", senderr)
```

File: plugins/outputs/timestream/timestream.go [hunk 6]
```diff
@@ -286,7 +286,7 @@ func (t *Timestream) writeToTimestream(writeRecordsInput *timestreamwrite.WriteR
 			t.logWriteToTimestreamError(notFound, writeRecordsInput.TableName)
 			// log error and return error to telegraf to retry in next flush interval
 			// We need this is to avoid data drop when there are no tables present in the database
-			return fmt.Errorf("failed to write to Timestream database '%s' table '%s', Error: '%s'",
+			return fmt.Errorf("failed to write to Timestream database '%s' table '%s', Error: '%w'",
```
Comment (Hipska) [inline plugins/outputs/timestream/timestream.go:289]:
```suggestion
			return fmt.Errorf("failed to write to Timestream database %q table %q: %w",
```
Same for the others in this file..

File: plugins/outputs/warp10/warp10.go [hunk 7]
```diff
@@ -122,7 +122,7 @@ func (w *Warp10) Write(metrics []telegraf.Metric) error {
 	addr := w.WarpURL + "/api/v0/update"
 	req, err := http.NewRequest("POST", addr, bytes.NewBufferString(payload))
 	if err != nil {
-		return fmt.Errorf("unable to create new request '%s': %s", addr, err)
+		return fmt.Errorf("unable to create new request '%s': %w", addr, err)
```
Comment (Hipska) [inline plugins/outputs/warp10/warp10.go:125]:
```suggestion
		return fmt.Errorf("unable to create new request %q: %w", addr, err)
```

**Review by Hipska — CHANGES_REQUESTED** [review]

File: plugins/outputs/azure_monitor/azure_monitor.go [hunk 8]
```diff
@@ -323,12 +324,13 @@ func (a *AzureMonitor) send(body []byte) error {
...
 	if err != nil {
-		return fmt.Errorf("unable to fetch authentication credentials: %v", err)
+		return fmt.Errorf("unable to fetch authentication credentials: %w", err)
 	}
 
 	resp, err := a.client.Do(req)
 	if err != nil {
-		if err.(*url.Error).Unwrap() == context.DeadlineExceeded {
+		var urlErr *url.Error
+		if errors.As(err, &urlErr) && errors.Is(urlErr, context.DeadlineExceeded) {
```
Comment (srebhan) [inline plugins/outputs/azure_monitor/azure_monitor.go:333]: I think this can be
```suggestion
		if errors.Is(urlErr, context.DeadlineExceeded) {
```
as the `errors.Is` function will unwrap until the target error-type is found...

File: plugins/outputs/dynatrace/dynatrace.go [hunk 9]
```diff
@@ -134,7 +134,7 @@ func (d *Dynatrace) Write(metrics []telegraf.Metric) error {
 		output := strings.Join(batch, "\n")
 		if output != "" {
 			if err := d.send(output); err != nil {
-				return fmt.Errorf("error processing data:, %s", err.Error())
+				return fmt.Errorf("error processing data:, %w", err)
```
[the remark this answers is not in the document]
Comment (srebhan) [reply plugins/outputs/dynatrace/dynatrace.go:137]: I agree to the @Hipska's suggestion. This is just ugly otherwise. Same below.

File: plugins/outputs/graylog/graylog.go [hunk 10]
```diff
@@ -457,9 +457,9 @@ func (g *Graylog) Write(metrics []telegraf.Metric) error {
 		}
 
 		for _, value := range values {
-			_, err := writer.Write([]byte(value))
+			_, err = writer.Write([]byte(value))
```
Comment (srebhan) [inline plugins/outputs/graylog/graylog.go:460]: Why this change? This increases the scope of `err`...

**Review by srebhan — COMMENTED** [review]
I do have a few minor comments. Thanks for this epic effort @zak-pawel!

File: plugins/outputs/azure_monitor/azure_monitor.go (hunk 8 above)
Comment (Hipska) [reply plugins/outputs/azure_monitor/azure_monitor.go:333]: Shouldn't it be `errors.Is(err, context.DeadlineExceeded)` in that case? As `urlErr` will always be `nil`?

Comment (zak-pawel) [reply plugins/outputs/azure_monitor/azure_monitor.go:333]: I would say:
`if errors.Is(err, context.DeadlineExceeded) {`

File: plugins/outputs/graylog/graylog.go (hunk 10 above)
Comment (zak-pawel) [reply plugins/outputs/graylog/graylog.go:460]: > Why this change? This increases the scope of `err`...

It doesn't matter here. I'm not a fan of shadowed variables - using them too often can lead to tricky errors.

**Review by srebhan — APPROVED** [review]
Huge thank you @zak-pawel!
